tools/format_all.sh

```diff
@@ -32,9 +32,9 @@ fi
 # Format the 100-line kernel demo as well
 KERNEL_DEMO_FILE="$WORKSPACE_ROOT/osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs"
 if [ "$CHECK_MODE" = true ]; then
-    cargo fmt --check -- $KERNEL_DEMO_FILE
+    rustfmt --check $KERNEL_DEMO_FILE
```

Comment:
The reason for this is that when I use `cargo fmt`, the changes made by `cargo fmt` are consistently rejected by `cargo fmt --check`.

In other words, after this PR, running
`cargo fmt -- osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs`
and then running
`cargo fmt --check -- osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs`
will always lead to a failure.

I am also unable to understand why this is happening, as this PR does not seem to change anything related to rustfmt.

Reply:
Wired, I can reproduce this, but... no idea why this happens:

```
root@localhost:~/asterinas# cargo fmt --check -- osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs
Diff in /root/asterinas/osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs:11:
 use alloc::sync::Arc;
 use alloc::vec;
 
-use ostd::arch::qemu::{exit_qemu, QemuExitCode};
+use ostd::arch::qemu::{QemuExitCode, exit_qemu};
 use ostd::cpu::UserContext;
 use ostd::mm::{
-    CachePolicy, FallibleVmRead, FallibleVmWrite, FrameAllocOptions, PageFlags, PageProperty,
-    Vaddr, VmIo, VmSpace, VmWriter, PAGE_SIZE,
+    CachePolicy, FallibleVmRead, FallibleVmWrite, FrameAllocOptions, PAGE_SIZE, PageFlags,
+    PageProperty, Vaddr, VmIo, VmSpace, VmWriter,
 };
 use ostd::prelude::*;
 use ostd::task::{Task, TaskOptions};
root@localhost:~/asterinas# cargo fmt -- osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs
root@localhost:~/asterinas# cargo fmt --check -- osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs
Diff in /root/asterinas/osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs:11:
 use alloc::sync::Arc;
 use alloc::vec;
 
-use ostd::arch::qemu::{QemuExitCode, exit_qemu};
+use ostd::arch::qemu::{exit_qemu, QemuExitCode};
 use ostd::cpu::UserContext;
 use ostd::mm::{
-    CachePolicy, FallibleVmRead, FallibleVmWrite, FrameAllocOptions, PAGE_SIZE, PageFlags,
-    PageProperty, Vaddr, VmIo, VmSpace, VmWriter,
+    CachePolicy, FallibleVmRead, FallibleVmWrite, FrameAllocOptions, PageFlags, PageProperty,
+    Vaddr, VmIo, VmSpace, VmWriter, PAGE_SIZE,
 };
 use ostd::prelude::*;
 use ostd::task::{Task, TaskOptions};
```

It first requires:
```diff
-use ostd::arch::qemu::{exit_qemu, QemuExitCode};
+use ostd::arch::qemu::{QemuExitCode, exit_qemu};
```

It then requires:
```diff
-use ostd::arch::qemu::{QemuExitCode, exit_qemu};
+use ostd::arch::qemu::{exit_qemu, QemuExitCode};
```

So no code can make `cargo fmt` happy: it does not know what it wants.

Reply:
Yes, I encounter the same issue locally. But using `rustfmt` instead of `cargo fmt` won't make things worse, so I just work around it.

Reply:
Thanks for the explanation. I don't have a better solution.

Since this only affects the format of the 100-line demo, switching to `rustfmt` may be acceptable.